controller/appcontroller.go

```diff
@@ -779,7 +779,13 @@ func (ctrl *ApplicationController) Run(ctx context.Context, statusProcessors int
 	if err != nil {
 		log.Warnf("Cannot init sharding. Error while querying clusters list from database: %v", err)
 	} else {
-		ctrl.clusterSharding.Init(clusters)
+		appItems, err := ctrl.getAppList(metav1.ListOptions{})
+
+		if err != nil {
+			log.Warnf("Cannot init sharding. Error while querying application list from database: %v", err)
```

Comment:
@jannfis 

Tracing through the code it looks like if the Cluster list is not populated, then apps would be handled by shard 0 or the first shard. 

If Init is never called, then updateDestribution is never called and the sharding cache's Shards map stays empty.

IsManagedCluster is used to determine whether an app/cluster should be managed by a particular shard and if it doesnt find a cluster in the Shards map, it returns shard-0.

Meaning if the Shards map stays empty, then basically its like sharding is off.

The one question I have is that the cluster cache has hooks for Add/Update/Delete events which would call the sharding cache's Add/Update/Delete events, which would call updateDestribution. So eventually the cluster list and the sharding cache will get a full list of the clusters and be able to give accurate sharding decisions.

But at the beginning it will just be shard-0 managing apps.

@akram Let me know if this is correct.

Reply:
@Enclavet yes, this is correct. If cluster's list cannot be populated then, then the behaviour will be similar to sharding not eanbled, and all clusters or apps will be processed by shard 0